packages/node-opentelemetry/src/spanProcessor.ts

```diff
@@ -1,11 +1,27 @@
 import { BatchSpanProcessor, Span } from '@opentelemetry/sdk-trace-base';
 import { Context } from '@opentelemetry/api';
+import { OTLPTraceExporter } from '@opentelemetry/exporter-trace-otlp-proto';
 
 import { hyperDXGlobalContext } from './context';
 
 export default class HyperDXSpanProcessor extends BatchSpanProcessor {
+  private readonly enableHDXGlobalContext: boolean;
+
+  constructor({
+    exporter,
+    enableHDXGlobalContext,
+  }: {
+    exporter: OTLPTraceExporter;
+    enableHDXGlobalContext: boolean;
+  }) {
+    super(exporter);
+    this.enableHDXGlobalContext = enableHDXGlobalContext;
+  }
+
   onStart(_span: Span, _parentContext: Context): void {
-    const traceId = _span.spanContext().traceId;
-    hyperDXGlobalContext.addTraceSpan(traceId, _span);
+    if (this.enableHDXGlobalContext) {
+      const traceId = _span.spanContext().traceId;
+      hyperDXGlobalContext.addTraceSpan(traceId, _span);
+    }
```

Comment:
are we sure we don't want to call the call the parent super.onStart(...) here? Is nothing missing if we don't from the default behavior?

Reply:
lol nvm 🤦 

![CleanShot 2024-05-21 at 11 57 53@2x](https://github.com/hyperdxio/hyperdx-js/assets/4743932/cdb1ff70-e118-4c14-89a6-1d0c6a72e800)